docs/shrc.md

```diff
@@ -22,33 +22,107 @@
+When a VM loads a class, the VM looks in the shared classes cache to see if the class is already present. If it is, it loads the class from the cache. Otherwise, it loads the class from the filesystem and writes it into the cache.
 
-You can treat the default cache like any other shared classes cache, for example you can print statistics for it, change the soft maximum limit size, or delete it.
+The VM detects filesystem updates by storing timestamp values into the cache and comparing the cached values with actual values on each class load operation. In this way, the VM detects when a class might be invalidated and can mark the class as *stale*. These operations happen transparently when classes are loaded, so users can modify and update as many classes as they like during the lifetime of a shared class cache, knowing that the correct classes are always loaded. Stale classes are *redeemed* if the same class is subsequently fetched by the class loader from another VM and checked against the stale class in the cache.
 
-You can enable class data sharing for non-bootstrap classes as well, by using `-Xshareclasses` without the `bootClassesOnly` suboption. You can also disable all class data sharing by using the `none` suboption.
+Occasionally, caches that are created from one version of the VM might not be compatible with caches that are created from a different version. This situation typically occurs when an update is made in OpenJ9 that changes the internal cache data structure. If a VM detects an incompatible cache at start up, it creates a new cache that can coexist, even if it has the same name. The VM detects a conflict by checking an internal shared classes cache generation number.
 
-## Best practices for using `-Xshareclasses`
+Caches are not compatible between VMs that are using different object storage modes. For example, a 64-bit VM that uses compressed references to store 64-bit objects in a 32-bit representation, cannot share a cache with a 64-bit VM that is not using compressed references. For more information about object storage options, see [Compressed references](allocation.md#compressed-references).
+
+In the OpenJ9 implementation of `java.net.URLClassLoader`, classes are read from and written to the cache using the public Helper API. Therefore, any class loader that extends `java.net.URLClassLoader` gets class sharing support for free. Custom  class loaders that do not extend `java.net.URLClassLoader` must be adapted to share class data as described in [Support for custom class loaders](#support-for-custom-class-loaders).
+
+### AOT code and JIT data
+
+OpenJ9 can automatically store small amounts of AOT code and JIT data, which helps improve performance in the following ways:
+
+- The VM dynamically compiles certain methods into AOT code at runtime. Subsequent VMs that attach to the cache can take advantage of the compiled code to start faster.
```

Comment:
Nitpick: It's the JIT compiler that "compiles certain methods into AOT", therefore I would change `VM` into `JIT compiler`